docs/architecture.md

```diff
@@ -150,6 +150,21 @@ if any write failed to one of the replicas, multiple differing chunk objects
 will be created in the backing store. See [Querier](#querier) for how data is
 deduplicated.
 
+#### Timestamp Ordering
+
+In general, all lines pushed to Loki for a given stream must have a newer
+timestamp than the line received before it. There are, however, two exceptions
+to this rule:
+
+1. If the incoming line exactly matches the previously received line (matching
+   both the previous timestamp and log text), the incoming line will be treated
+   as a duplicate log and ignored. To avoid having this happen, logs should be
+   pushed to Loki with at least millisecond precision.
```

Comment:
I think maybe alter this to say `... treated as an exact duplicate and ignored.` and then remove the last sentence completely.